### PR TITLE
Fix/storage ef test

### LINKED ISSUE #761: bug: setup_test_state error for storage
### Current behavior

The storage var for `storage_` in contracts are never set.

https://github.com/kkrt-labs/kakarot/blob/b41e2590bb4eadbd027cff9d02a2acc22c38c6c8/tests/integration/ef_tests/test_ef_blockchain_tests.py#L41-L52
https://github.com/kkrt-labs/kakarot/blob/b41e2590bb4eadbd027cff9d02a2acc22c38c6c8/tests/integration/ef_tests/test_ef_blockchain_tests.py#L53-L56

https://github.com/kkrt-labs/kakarot/blob/b41e2590bb4eadbd027cff9d02a2acc22c38c6c8/tests/integration/ef_tests/test_ef_blockchain_tests.py#L81-L85

### Expected behavior
The `storage_` var should be set to the actual storage of the contract.

### Code
The issue can be done by following this as a helper:
https://github.com/jobez/kakarot/blob/67dd75aac0059b7ec19c34116fb247441ed979ed/tests/integration/ef_tests/state_management.py#L60

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR: 0.1 day

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #761

### CODE REVIEW

File: tests/integration/ef_tests/test_ef_blockchain_tests.py [hunk 1]
```diff
@@ -79,11 +79,67 @@ async def setup_test_state(
...
+
+                # Now we have to perform low level custom storage for Starknet
+                # Reference: https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/contract-storage/
+                for storage_key, storage_value in ca_storage_entries:
+                    # The address for storage is pedersen(<STORAGE_VAR_NAME>, key_1, key_2)
+                    # Here: pedersen("storage_", high, low)
+                    await starknet.state.state.set_storage_at(
+                        contract_address=starknet_address,
+                        key=get_storage_var_address(
+                            "storage_", storage_key[0], storage_key[1]
```
Comment (greged93) [inline tests/integration/ef_tests/test_ef_blockchain_tests.py:125]: can you do storage_key... here?

File: tests/integration/ef_tests/test_ef_blockchain_tests.py [hunk 2]
```diff
@@ -79,11 +79,67 @@ async def setup_test_state(
...
+                        ),
+                        value=storage_value[0],
+                    )
+
+                    # Since we need to store both a high and low element, the second value to store is simply located
+                    # At the address calculated above + 1
+                    # We're able to store 256 felts this way in a custom Struct.
+                    # Above, we'll need to compute a new address to prevent collision
+                    await starknet.state.state.set_storage_at(
```
Comment (greged93) [inline tests/integration/ef_tests/test_ef_blockchain_tests.py:134]: Given that we do twice the same action, we could just iterate on storage_value no?

File: tests/integration/ef_tests/test_ef_blockchain_tests.py [hunk 3]
```diff
@@ -79,11 +79,67 @@ async def setup_test_state(
...
+                    # We append to the storage entries a tuple of (key: Cairo.U256, value: Cairo.U256)
+                    # Note that when serializing, we need to store low first and then high:
+                    # {low: u128, high: 128} => serde => [low, high]
+                    ca_storage_entries.append(
+                        ([key_low, key_high], [value_low, value_high])
+                    )
+
+                # Now we have to perform low level custom storage for Starknet
+                # Reference: https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/contract-storage/
+                for storage_key, storage_value in ca_storage_entries:
```
Comment (greged93) [inline tests/integration/ef_tests/test_ef_blockchain_tests.py:119]: do we need to reiterate on the `ca_storage_entries`? I feel like all this loop could be brought up to the `for index in range(len(storage_keys)):` loop.

**Review by greged93 — COMMENTED** [review]
nice fix thanks!

File: tests/integration/ef_tests/test_ef_blockchain_tests.py (hunk 1 above)
Comment (Eikix) [reply tests/integration/ef_tests/test_ef_blockchain_tests.py:125]: what is this?

**Comment by Eikix** [comment]
Checked locally and this seems to work

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**